view/frontend/web/js/view/payment/method-renderer/adyen-hpp-method.js

```diff
@@ -344,22 +346,49 @@ define(
-
-                                    if ('isAvailable' in component) {
-                                        component.isAvailable().then(() => {
-                                            component.mount(containerId);
-                                        }).catch(e => {
-                                            result.isAvailable(false);
-                                        });
+                                    var url = new URL(location.href);
+                                    if (
+                                        paymentMethod.methodIdentifier === 'amazonpay'
+                                        && url.searchParams.has(amazonSessionKey)
+                                    ) {
+                                        configuration = {
+                                            amazonCheckoutSessionId: url.searchParams.get(amazonSessionKey),
+                                            showOrderButton: false,
+                                            amount: {
+                                                currency: configuration.amount.currency,
+                                                value: configuration.amount.value
+                                            },
+                                            returnUrl: location.href,
+                                            showChangePaymentDetailsButton: false
+                                        }
+                                        const component = self.checkoutComponent.create(
+                                            paymentMethod.methodIdentifier, configuration);
+                                        const containerId = '#adyen-alternative-payment-container-' +
+                                            paymentMethod.methodIdentifier;
+                                        component.mount(containerId).submit();
                                     } else {
-                                        component.mount(containerId);
+                                        const component = self.checkoutComponent.create(
+                                            paymentMethod.methodIdentifier, configuration);
+                                        const containerId = '#adyen-alternative-payment-container-' +
+                                            paymentMethod.methodIdentifier;
+
+                                        if ('isAvailable' in component) {
+                                            component.isAvailable().then(() => {
+                                                component.mount(containerId);
+                                            }).catch(e => {
+                                                result.isAvailable(false);
+                                            });
+                                        } else {
+                                            component.mount(containerId);
+                                        }
```

Comment:
I think this can be refactored so that there's only one .create and containerId declaration, etc. but can't think of an alternative right now